Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -1418,10 +1419,7 @@ namespace AzToolsFramework
                     parentEntitiesToUpdate.insert(GetEntityById(parentEntityId));
                 }
 
-                commonOwningInstance->get().DetachEntity(entityId).release();
-                AZ::ComponentApplicationBus::Broadcast(&AZ::ComponentApplicationRequests::DeleteEntity, entityId);
-
-                detachedEntityAliasPaths.push_back(AZStd::move(nestedEntityAliasPath));
+                commonOwningInstance->get().DetachEntity(entityId).reset();
```

Comment:
nit: Don't need to explicitly reset the entity as it'll be destroyed at the end of this scope